github/projects.go

```diff
@@ -0,0 +1,218 @@
+	Before string `url:"before,omitempty"`
+
+	// A cursor, as given in the Link header. If specified, the query only searches for events after this cursor.
+	After string `url:"after,omitempty"`
+
+	// For paginated result sets, the number of results to include per page.
+	PerPage int `url:"per_page,omitempty"`
+}
+
+// ListProjectsOptions specifies optional parameters to list projects for user / organization.
+type ListProjectsOptions struct {
+	ListProjectsPaginationOptions
+
+	// Q is an optional query string to limit results to projects of the specified type.
+	Query string `url:"q,omitempty"`
+}
+
+// ProjectV2FieldOption represents an option for a project field of type single_select or multi_select.
+// It defines the available choices that can be selected for dropdown-style fields.
+//
+// GitHub API docs: https://docs.github.com/rest/projects/fields
+type ProjectV2FieldOption struct {
+	ID string `json:"id,omitempty"`
+	// The display name of the option.
+	Name string `json:"name,omitempty"`
+	// The color associated with this option (e.g., "blue", "red").
+	Color string `json:"color,omitempty"`
+	// An optional description for this option.
+	Description string `json:"description,omitempty"`
+}
+
+// ProjectV2Field represents a field in a GitHub Projects V2 project.
+// Fields define the structure and data types for project items.
+//
+// GitHub API docs: https://docs.github.com/rest/projects/fields
+type ProjectV2Field struct {
+	ID        *int64     `json:"id,omitempty"`
+	NodeID    string     `json:"node_id,omitempty"`
+	Name      string     `json:"name,omitempty"`
+	DataType  string     `json:"dataType,omitempty"`
+	URL       string     `json:"url,omitempty"`
+	Options   []*any     `json:"options,omitempty"`
+	CreatedAt *Timestamp `json:"created_at,omitempty"`
+	UpdatedAt *Timestamp `json:"updated_at,omitempty"`
+}
+
+// ListProjectsForOrg lists Projects V2 for an organization.
+//
+// GitHub API docs: https://docs.github.com/rest/projects/projects#list-projects-for-organization
+//
+//meta:operation GET /orgs/{org}/projectsV2
+func (s *ProjectsService) ListProjectsForOrg(ctx context.Context, org string, opts *ListProjectsOptions) ([]*ProjectV2, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/projectsV2", org)
+	u, err := addOptions(u, opts)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var projects []*ProjectV2
+	resp, err := s.client.Do(ctx, req, &projects)
+	if err != nil {
+		return nil, resp, err
+	}
+	return projects, resp, nil
+}
+
+// GetProjectForOrg gets a Projects V2 project for an organization by ID.
+//
+// GitHub API docs: https://docs.github.com/rest/projects/projects#get-project-for-organization
+//
+//meta:operation GET /orgs/{org}/projectsV2/{project_number}
+func (s *ProjectsService) GetProjectForOrg(ctx context.Context, org string, projectNumber int64) (*ProjectV2, *Response, error) {
```

Comment:
Perhaps the type of `projectNumber` should be the same as the type in the `ProjectV2` structure:


```suggestion
func (s *ProjectsService) GetProjectForOrg(ctx context.Context, org string, projectNumber int) (*ProjectV2, *Response, error) {
```

Reply:
Great catch, @alexandear!  This made me curious if the `Number` property ever might be as huge as an "id" and as far as I can tell, that is not the case, so I do _NOT_ think both should be `int64`, but instead both should be `int` as you have suggested.

Thoughts, @JoannaaKL?

Reply:
In the GH API project's ID is a `number` type and `project_number` is an integer type. It can hold int64 actually. `number` type can hold any number so yes, I think it should be equal to int 64 as it is now. So I don't think they should be the same.

Reply:
I was talking about types described in the doc https://docs.github.com/en/rest/projects/fields?apiVersion=2022-11-28
`projectNumber` there is described as `integer` and that type can hold int64 [link](https://swagger.io/docs/specification/v3_0/data-models/data-types/#numbers) (correct me if I am wrong).